### PR TITLE
add toggle to controle inventory parse as error

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
also rearranged new inventory options into it's own ini section

##### ISSUE TYPE

 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME

inventory
##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

**Comment by AlanCoding** [comment]
I checked out the branch and I'm trying to test with `ansible-inventory`, but maybe I'm doing something wrong. Still giving 0 error code here, giving it an exploding script that raises an exception:

```
$ ansible-inventory -i ~/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py --list
 [WARNING]:  * Failed to parse /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py with script plugin: problem
running /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py --list ([Errno 2] No such file or directory)

 [WARNING]:  * Failed to parse /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py with ini plugin:
/Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py:3: Expected key=value host variable assignment, got:
Exception(this is a python exception)

 [WARNING]: Unable to parse /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: Could not match supplied host pattern, ignoring: all

{
    "_meta": {
        "hostvars": {}
    }, 
    "all": {
        "children": [
            "ungrouped"
        ]
    }, 
    "ungrouped": {}
}
(ansible)arominge-OSX:ansible alancoding$ echo $?
0
```

within the directory, I have a fresh `.ansible.cfg` with `unparsed_is_failed=True`. I install in a virutalenv with `python setup.py install`. I have also been trying to use as an environment variable with

**Comment by bcoca** [comment]
@AlanCoding weird:
```
#>ANSIBLE_INVENTORY_UNPARSED_FAILED=1 ansible-inventory --graph
```
gets me:
```
ERROR! Could not parse inventory source /etc/ansible/hosts/z_test.config with availabel plugins:
Plugin ini failed: /etc/ansible/hosts/z_test.config:1: Expected key=value host variable assignment, got: constructed
```
and without:
```
#>ansible-inventory --graph                                    
```
I get:
```
 [WARNING]:  * Failed to parse /etc/ansible/hosts/z_test.config with ini plugin: /etc/ansible/hosts/z_test.config:1: Expected key=value host variable
assignment, got: constructed

 [WARNING]: Unable to parse /etc/ansible/hosts/z_test.config as an inventory source
...
```

Have you put the setting in an `[inventory]` section?

**Comment by AlanCoding** [comment]
Sorry, I had to start over with my install. Using the environment variable is giving the desired results.

```
$ ANSIBLE_INVENTORY_UNPARSED_FAILED=1 ansible-inventory -i ~/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py --list
ERROR! Could not parse inventory source /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py with availabel plugins:
Plugin script failed: problem running /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py --list ([Errno 2] No such file or directory)
Plugin ini failed: /Users/alancoding/Documents/repos/ansible-inventory-file-examples/not_valid/invalid_script.py:3: Expected key=value host variable assignment, got: Exception(this is a python exception)

$ echo $?
4
```

The exploding script example is correctly failing +1

File: lib/ansible/inventory/manager.py [hunk 1]
```diff
@@ -267,9 +267,17 @@ def parse_source(self, source, cache=True):
             else:
                 if not parsed and failures:
                     # only if no plugin processed files should we show errors.
-                    for fail in failures:
-                        display.warning('\n* Failed to parse %s with %s plugin: %s' % (to_native(fail['src']), fail['plugin'], to_native(fail['exc'])))
-                        display.vvv(fail['exc'].tb)
+                    if C.INVENTORY_UNPARSED_IS_FAILED:
+                        msg = "Could not parse inventory source %s with availabel plugins:\n" % source
```
Comment (simfarm) [inline lib/ansible/inventory/manager.py:271]: @bcoca Typo: "availabel."

**Pull request merged**